### ISSUE TITLE
Bad quote escape with "git-bash" (windows)

### ISSUE THREAD

**Issue by TotoLastro** [evilmartians/lefthook#269]
Hi all,

After installing lefthook from npm, I did some tests.
And I see a problem with quotes (a bad escape ?).

For the example, this is the lefthook.yml :
```yaml
pre-commit:
  parallel: false
  commands:
    echo-files:
      run: echo "{staged_files}"
```
And this is the result after running the command :
```
$ lefthook -v run pre-commit
RUNNING HOOKS GROUP: pre-commit
  waiting Files before filters:
 [lefthook.yml]
Files after filters:
 [lefthook.yml]
Files after escaping:
 [lefthook.yml]

  EXECUTE > echo-files
\lefthook.yml"


SUMMARY: (done in 0.17 seconds)
```
(we can see trailing chars : **\\** and **"**)

No problem if I use apostrophes (the **echo** command prints only the filename)

- lefthook 0.7.7
- Git Bash (x64) 2.36.0.1
- Windows 10 Enterprise N 20H2

**Comment by mrexox** [comment]
Hey! We've released new version of lefthook recently. Could you check if there's still a bug with the latest lefthook binary?

**Comment by Carduelis** [comment]
@mrexox Unfortunately, it doesn't work for me.
I'm using next.js and there is a common pattern to have the `[slug].tsx` filenames.
I'm on Windows and if I run the command with double quotes, it works:
```
yarn eslint "pages/blog/[slug].tsx"
```

But the lefthook uses single quotes which leads to the following error:
```
Oops! Something went wrong! :(

ESLint: 8.17.0

No files matching the pattern "'pages/blog/[slug].tsx'" were found.
Please check for typing mistakes in the pattern.
```

**Comment by Carduelis** [comment]
My `lefthook.yml` is:
```
pre-commit:
  parallel: true
  commands:
    prettier:
      glob: '*.{js,ts,jsx,tsx}'
      run: yarn prettier --write {staged_files} && git add {staged_files}
    eslint:
      glob: '*.{js,ts,jsx,tsx}'
      run: yarn eslint {staged_files} --fix && git add {staged_files}
```

How can I replace `'` with `"`?

**Comment by Carduelis** [comment]
For those, who experience issues using next.js framework with [Dynamic Routes](https://nextjs.org/docs/routing/dynamic-routes), there is a temporal solution to ignoring files for optional wildcard routing.

If you use Typescript and next.js you can use the following regexp: `\[(\[)?(\.\.\.)?\w+\](\])?.tsx`
```
### Legend for regexp
### \[(\[)?     [ or [[
### (\.\.\.)?   optional dots ...
### \w          slug name
### \](\])?     ] or ]]
### .tsx        we use only Typescript
```

Which will exclude from the check those files:
```
[slug].tsx
[[slug]].tsx
[...slug].tsx
[[...slug]].tsx
```

It will

**Comment by mrexox** [comment]
Hey, @TotoLastro , @Carduelis ! I have prepared a fix: https://github.com/evilmartians/lefthook/pull/314. I hope it will fit in your cases but if you have time and will you can pull the code, build a `lefthook` executable, and test it.

Or if you have a public repo with the appropriate files and lefthook.yml, just send me a link, and I will test it myself.

I am going to release this fix with version 1.1.0 in about a week.
